src/cets.erl

```diff
@@ -0,0 +1,420 @@
+    cets_call:sync_operation(Server, {insert, Rec}).
+
+-spec insert_many(server_ref(), list(tuple())) -> ok.
+insert_many(Server, Records) when is_list(Records) ->
+    cets_call:sync_operation(Server, {insert_many, Records}).
+
+%% Removes an object with the key from all nodes in the cluster.
+%% Ideally, nodes should only remove data that they've inserted, not data from other node.
+-spec delete(server_ref(), term()) -> ok.
+delete(Server, Key) ->
+    cets_call:sync_operation(Server, {delete, Key}).
+
+%% A separate function for multidelete (because key COULD be a list, so no confusion)
+-spec delete_many(server_ref(), [term()]) -> ok.
+delete_many(Server, Keys) ->
+    cets_call:sync_operation(Server, {delete_many, Keys}).
+
+-spec insert_request(server_ref(), tuple()) -> request_id().
+insert_request(Server, Rec) ->
+    cets_call:async_operation(Server, {insert, Rec}).
+
+-spec insert_many_request(server_ref(), [tuple()]) -> request_id().
+insert_many_request(Server, Records) ->
+    cets_call:async_operation(Server, {insert_many, Records}).
+
+-spec delete_request(server_ref(), term()) -> request_id().
+delete_request(Server, Key) ->
+    cets_call:async_operation(Server, {delete, Key}).
+
+-spec delete_many_request(server_ref(), [term()]) -> request_id().
+delete_many_request(Server, Keys) ->
+    cets_call:async_operation(Server, {delete_many, Keys}).
+
+-spec wait_response(request_id(), non_neg_integer() | infinity) -> ok.
+wait_response(Mon, Timeout) ->
+    cets_call:wait_response(Mon, Timeout).
+
+%% Get a list of other CETS processes that are handling this table.
+-spec other_servers(server_ref()) -> [server_ref()].
+other_servers(Server) ->
+    cets_call:long_call(Server, other_servers).
+
+%% Get a list of other nodes in the cluster that are connected together.
+-spec other_nodes(server_ref()) -> [node()].
+other_nodes(Server) ->
+    lists:usort(pids_to_nodes(other_pids(Server))).
+
+%% Get a list of other CETS processes that are handling this table.
+-spec other_pids(server_ref()) -> [pid()].
+other_pids(Server) ->
+    other_servers(Server).
+
+-spec pause(server_ref()) -> pause_monitor().
+pause(Server) ->
+    cets_call:long_call(Server, pause).
+
+-spec unpause(server_ref(), pause_monitor()) -> ok | {error, unknown_pause_monitor}.
+unpause(Server, PauseRef) ->
+    cets_call:long_call(Server, {unpause, PauseRef}).
+
+%% Waits till all pending operations are applied.
+-spec sync(server_ref()) -> ok.
+sync(Server) ->
+    cets_call:long_call(Server, sync).
+
+-spec ping(server_ref()) -> pong.
+ping(Server) ->
+    cets_call:long_call(Server, ping).
+
+-spec info(server_ref()) -> info().
+info(Server) ->
+    cets_call:long_call(Server, get_info).
+
+%% gen_server callbacks
+
+-spec init({table_name(), start_opts()}) -> {ok, state()}.
+init({Tab, Opts}) ->
+    process_flag(message_queue_data, off_heap),
+    MonTab = list_to_atom(atom_to_list(Tab) ++ "_mon"),
+    _ = ets:new(Tab, [ordered_set, named_table, public]),
```

Comment:
Another note: why `ordered_set`? It slows down read time. Also: how about allowing some customisation instead of hardcoding the table type?

Reply:
ordered_set is required for ejabberd_sm to work - for reads all sessions of the user, for example.